### PR TITLE
Add information about compressed refs and SOA/LOA

### PR DESCRIPTION
**Pull request by SueChaplain**
- add to memory allocation topic
- link to -Xmx topic for large page guidance
- update -Xmx to add guidance for large pages

Signed-off-by: SueChaplain <sue_chaplain@uk.ibm.com>

### CODE REVIEW

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by SueChaplain** [comment]
@dmitripivkine - can you help out here please. I've overwhelmed Aleks with other stuff. The LOA/SOA content is already reviewed/approved, but the compressed refs content needs checking. Staged at https://eclipse.github.io/openj9-docs-staging/683/allocator/

**Comment by SueChaplain** [comment]
Jenkins doc stage

File: docs/allocator.md [hunk 1]
```diff
@@ -65,5 +65,34 @@ The following options control the requested TLH size:
...
+
+Compressed references are used by default when the maximum Java heap size is in the range 0 - 57 GB on AIX&reg;, Linux&reg;, and Windows&reg; systems. The upper limit is also 57 GB on z/OS&reg; systems that have APAR OA49416
+installed (25 GB without APAR OA49416). All GC policies observe these limits except for the [`metronome`](gc.md#metronome-policy) policy, which can only support a heap size of up to 25 GB with compressed references.
+
+When the VM uses compressed references, classes, threads, and monitors are stored in the lowest 4 GB of address space. However, this area of memory is also used by native libraries, the operating system, and for small Java heaps. If you receive native memory `OutOfMemoryError` exceptions when running with compressed references enabled, these errors might result from the lowest 4 GB of address space becoming full. Try specifying a large heap with the [`-Xmx`](xms.md) option, which puts the Java heap into a higher area of address space or using the [`-Xmcrs`](xmcrs.md) option to reserve space in the lowest 4 GB of address space for compressed references.
```
Comment (dmitripivkine) [inline docs/allocator.md:92]: I believe we can be more specific regarding OOM type: "out of native memory below 4GB bar". If we have description of possible types of OOM the wording should match obviously.

File: docs/allocator.md [hunk 2]
```diff
@@ -65,5 +65,34 @@ The following options control the requested TLH size:
...
+
+The GC action that is triggered by the allocation failure depends on the GC policy in force.
+
+The overall size of the LOA is calculated when the heap is initialized, and recalculated at the end of each global GC cycle. The GC can expand or shrink the LOA, depending on usage, to avoid allocation failures.
+
+You can control the size of the LOA by using the `-Xloainitial`, `-Xloaminimum`, and `-Xloamaximum` command line options (See [LOA sizing options](xloaminimum.md)). If the LOA is not used, the GC shrinks the LOA after a few cycles, down to the value of `-Xloaminimum`. You can also specify [`-Xnoloa`](xloa.md) to prevent an LOA being created.
+
+An SOA and LOA are used by the OpenJ9 standard GC policies: `gencon`, `optavgpause` and `optthruput`. However, the `balanced` and `metronome` GC policies use a heap configuration that does not include the SOA or LOA. For more information about policies, see [Garbage collection policies](gc.md).
 
+### Compressed references
```
Comment (dmitripivkine) [inline docs/allocator.md:85]: Please be aware that this paragraph might require editing in nearest future with introduction of Compressed/Full 64-bit Mixed build (planned for 0.25 release). This single build is going to replace current "Compressed" and "Large Heap" builds. The running mode "Compressed" or "Full 64-bit" will be selected automatically based on provided command line options.

File: docs/xms.md [hunk 3]
```diff
@@ -51,6 +49,13 @@ If the scavenger is disabled, `-Xms` &ge; `-Xmo`
 See [Using -X command-line options](x_jvm_commands.md) for more information about the `<size>` parameter.  
 See [Default settings for the OpenJ9 VM](openj9_defaults.md) for more about default values.
 
+### Calculating `-Xmx` for large pages
+
+For z/OS&reg; and Linux&reg; on IBM Z&reg;, specifying an object heap size that is a multiple of the page size uses another page of memory. For example, if the page size is 2 GB, setting `-Xmx2G` actually uses 4 GB of memory. To avoid using more memory, make the heap size a little smaller than an integral number of pages by subtracting at least 16 bytes from the value that you set with the `-Xmx` option. The following examples are provided:
```
Comment (dmitripivkine) [inline docs/xms.md:54]: I am pretty sure this is obsolete. The 8 bytes tail padding on z platforms has been removed

Comment (dmitripivkine) [reply docs/xms.md:54]: FYI Object Heap Tail Padding (allocation slightly more memory than requested) is done by performance improvement reason if `-Xaggressive` option is provided for all platforms. If size of physical page is smaller than 1GB one extra memory page is allocated. If size of physical page is equal or larger than 1GB no extra memory is allocated.

**Comment by dmitripivkine** [comment]
In general may be make sense to add examples how `-Xmcrs` can be used to avoid native memory low 4GB bar OOM. It is confusing subject for customers

File: docs/xms.md (hunk 3 above)
Comment (SueChaplain) [reply docs/xms.md:54]: >I am pretty sure this is obsolete.

Yes you are right, I didn't spot the tagging. Removing this.

File: docs/allocator.md (hunk 2 above)
Comment (SueChaplain) [reply docs/allocator.md:85]: Noted @dmitripivkine - thanks. I guess we will have a docs issue for this to ensure it isn't missed.

**Comment by SueChaplain** [comment]
@pmhayward - the changed content here has now been reviewed (SOA/LOA by Aleks, compressed refs by Dmitri). Please can you take a quick look and merge if you are happy. There are further structural changes I hope to put in before 0.24, so would like to proceed with these. Thanks.

**Comment by SueChaplain** [comment]
Jenkins doc stage

**Comment by pmhayward** [comment]
LGTM

**Pull request merged**